internal/build/build.go

```diff
@@ -429,7 +437,62 @@ func buildAllPkgs(ctx *context, initial []*packages.Package, verbose bool) (pkgs
 	return
 }
 
-func linkMainPkg(ctx *context, pkg *packages.Package, pkgs []*aPackage, conf *Config, mode Mode, verbose bool) {
+var (
+	globalNames = make(map[string][]string)
+	globalDatas = make(map[string]string)
+	errXflags   = errors.New("-X flag requires argument of the form importpath.name=value")
+)
+
+func addGlobalString(arg string, mainPkgs []string) {
+	eq := strings.Index(arg, "=")
+	dot := strings.LastIndex(arg[:eq+1], ".")
+	if eq < 0 || dot < 0 {
+		panic(errXflags)
+	}
+	pkg := arg[:dot]
+	pkgs := []string{pkg}
+	if pkg == "main" {
+		pkgs = mainPkgs
+	}
+	for _, pkg := range pkgs {
+		name := pkg + arg[dot:eq]
+		value := arg[eq+1:]
+		if _, ok := globalDatas[name]; !ok {
+			globalNames[pkg] = append(globalNames[pkg], name)
+		}
+		globalDatas[name] = value
+	}
+}
+
+func createGlobals(prog llssa.Program, pkgs []*aPackage) (string, error) {
+	if len(globalDatas) == 0 {
+		return "", nil
+	}
+	for _, pkg := range pkgs {
+		if pkg.ExportFile == "" {
+			continue
+		}
+		if names, ok := globalNames[pkg.PkgPath]; ok {
+			err := llssa.UndefinedGlobalStrings(pkg.LPkg, names)
+			if err != nil {
+				return "", err
+			}
+			pkg.ExportFile = pkg.ExportFile + "-global.ll"
+			os.WriteFile(pkg.ExportFile, []byte(pkg.LPkg.String()), 0644)
+		}
+	}
+	global := prog.NewPackage("", "global")
+	llssa.AddGlobalStrings(global, globalDatas)
+	f, err := os.CreateTemp("", "global*.ll")
```

Comment:
Is it possible to do it without creating a temporary file?

Reply:
change to make pkg.ExportFile+"-global.ll"